### PR TITLE
Cancel API requests enhancement

### PR DESCRIPTION
**Pull request by debbieellis7**
Resolved: #77

### CODE REVIEW

File: assets/react/reducers/cancelApiRequestsReducer.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        ...state,
+        abortControllers: list
+      }
+    }
+
+    case CANCEL_REQUESTS: {
+      const abortControllers = [...state.abortControllers]
+      const list = []
+
+      abortControllers.map(item => list.push(item.abort()))
```
Comment (jakejackson1) [inline assets/react/reducers/cancelApiRequestsReducer.js:24]: This is a side effect. Needs to be in a Saga, not in the reducer.

File: assets/react/sagas/pdf.js [hunk 2]
```diff
@@ -195,19 +204,8 @@ export function * generatePdf (pdfs) {
...
-  const data = {
-    pdf,
-    signal: abortController.signal
-  }
-
-  yield put({ type: STORE_ABORT_CONTROLLER, payload: abortController })
-
   try {
-    const response = yield retry(3, 1000, apiRequestGeneratePdf, data)
+    const response = yield retry(3, 1000, apiRequestGeneratePdf, pdf)
```
Comment (jakejackson1) [inline assets/react/sagas/pdf.js:208]: Why is the PDF request no longer abortable?

File: assets/react/sagas/form.js [hunk 3]
```diff
@@ -51,8 +52,22 @@ export function * watchGetSelectedEntriesId () {
  * @since 1.0
  */
 export function * getSelectedEntriesId (payload) {
+  /* Initialize AbortController */
+  const abortController = new window.AbortController()
```
Comment (jakejackson1) [inline assets/react/sagas/form.js:56]: We're just creating more work for ourselves by doing this manually for each API request. This should be done directly in the `api()` function call to support all API requests automatically.

File: assets/react/api/form.js [hunk 4]
```diff
@@ -18,11 +18,12 @@ import { api } from './api'
  *
  * @since 1.0
  */
-export const apiRequestAllEntryIds = ({ formId, filterData }) => {
+export const apiRequestAllEntryIds = ({ formId, filterData, signal }) => {
   const url = `${GPDF_BULK_GENERATOR.restUrl}/search/${formId}/entries`
 
   return api(url, {
     method: 'POST',
+    signal,
```
Comment (jakejackson1) [inline assets/react/api/form.js:26]: We're just creating more work for ourselves by passing signal around manually. We should create the signature directly in the `api()` function instead.

File: assets/react/sagas/pdf.js (hunk 2 above)
Comment (debbieellis7) [reply assets/react/sagas/pdf.js:195]: We're going to make full use of the 'cancel' effect from redux-saga @jakejackson1 .

**Pull request merged**